### DISCUSSION TITLE
Are there any leaks if VPN connection is lost?

### DISCUSSION THREAD

**Discussion by mitohund** [qdm12/gluetun#868] [Q&A] [answered]
Quick praise: gluetun is super impressive. I'm relatively new to docker and self-hosting. But this is one of the most polished, well-documented and easy to use projects I've come across. I set it up and it's working flawlessly so far. Kudos!

My question: let's say the connection to the VPN is lost or some other error happens, is there any possibility of information leaking? I'm guessing there isn't, since all traffic goes through the gluetun container and I'm sure that one is set up to not leak any information. But if I read a line somewhere confirming this, my mind would be a lot more at ease.

Maybe it would also make sense to add a line explaining this to the main readme? I'm sure I'm not the only noob-ish person wondering about this.

**Comment by qdm12** [comment, marked as answer]
> Quick praise

Praise appreciated thanks!!

> But if I read a line somewhere confirming this, my mind would be a lot more at ease.

Yes the firewall takes care to only allow inbound from your lan and outbound is allowed only though the vpn network interface, at the vpn server ip and at the vpn server port. This gets enabled about 15ms after container start (can't do faster for now) and gets removed before container shutdown only.

I should probably add this to the wiki somewhere 😄

> Maybe it would also make sense to add a line explaining this to the main readme?

There are too many features in the readme haha, but it's already in there (with a grammar/typo I just spotted):

> Built in firewall kill switch to allow traffic only with needed the VPN servers and LAN devices

**Reply by mitohund** [reply]
Fantastic, thank you for the quick reply and for clarifying things for me. I'll now dive into it a bit more and start using it more extensively. Thanks again!
